packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -74,12 +85,13 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
       })
     }
 
-    if (channel) {
-      channel.onmessage = (event) => {
-        watchPaused = true
-        cookies[name] = cookie.value = opts.decode(event.data)
-        nextTick(() => { watchPaused = false })
+    if (store) {
+      store.onchange = (event) => {
+        const cookie = event.changed.find((c: any) => c.name === name)
+        if (cookie) handleChange({ value: cookie.value })
```

Comment:
Would we add `klona` copy here as well, since we add raw value to cookies[name] here basically? Matching issue in https://github.com/nuxt/nuxt/pull/25255

Reply:
Oops, I didn't notice you don't use cookie from line 56 and this one is from separate event. Doesn't usually see same variables name per scope :D 

Thanks for merging main btw